load.c

```diff
@@ -1511,6 +1515,29 @@ rb_f_autoload_p(int argc, VALUE *argv, VALUE obj)
     return rb_mod_autoload_p(argc, argv, klass);
 }
 
+void *
+rb_dln_resolve_symbol(const char* fname, const char* symbol)
+{
+    VALUE handle;
+    VALUE resolved;
+    VALUE path;
+    VALUE fname_str = rb_str_new_cstr(fname);
+
+    resolved = rb_resolve_feature_path((VALUE)NULL, fname_str);
+    if (NIL_P(resolved) || RARRAY_LEN(resolved) != 2) {
+        return NULL;
+    }
```

Comment:
You might need to consider statically linked extensions here.

Reply:
I updated the code on the assumption that the feature name of statically linked extensions are also included in `$LOADED_FEATURES`. Is that true?

Reply:
You can use `rb_feature_p()` instead, I guess.